app/src/main/java/org/glucosio/android/presenter/ExportPresenter.java

```diff
@@ -56,84 +56,78 @@
     private int toDay;
     private int toMonth;
     private int toYear;
-    private DatabaseHandler dB;
-    private MainActivity activity;
-
-    public ExportPresenter(MainActivity exportActivity, DatabaseHandler dbHandler) {
-        this.activity = exportActivity;
-        dB = dbHandler;
-    }
 
-    public void onExportClicked(final Boolean all) {
-        final List<GlucoseReading> readings;
+    private Activity mActivity;
+    private ExportView mExportView;
+    private DatabaseHandler dB;
 
-        if (all) {
-            readings = dB.getGlucoseReadings();
+    public ExportPresenter(Activity activity) {
```

Comment:
Because you need the Activity since you are handling the permissions in the presenter (it wouldn't be really my choice but see line 140) which requires to have an activity, the cast is to ensure said activity is implementing the required View interface.